azure/scope/machine.go

```diff
@@ -228,22 +228,18 @@ func (m *MachineScope) InboundNatSpecs() []azure.ResourceSpecGetter {
 
 // NICSpecs returns the network interface specs.
 func (m *MachineScope) NICSpecs() []azure.ResourceSpecGetter {
-	spec := &networkinterfaces.NICSpec{
-		Name:                  azure.GenerateNICName(m.Name()),
-		ResourceGroup:         m.ResourceGroup(),
-		Location:              m.Location(),
-		SubscriptionID:        m.SubscriptionID(),
-		MachineName:           m.Name(),
-		VNetName:              m.Vnet().Name,
-		VNetResourceGroup:     m.Vnet().ResourceGroup,
-		SubnetName:            m.AzureMachine.Spec.SubnetName,
-		AcceleratedNetworking: m.AzureMachine.Spec.AcceleratedNetworking,
-		DNSServers:            m.AzureMachine.Spec.DNSServers,
-		IPv6Enabled:           m.IsIPv6Enabled(),
-		EnableIPForwarding:    m.AzureMachine.Spec.EnableIPForwarding,
-		AdditionalTags:        m.AdditionalTags(),
-		ClusterName:           m.ClusterName(),
+	nicSpecs := []azure.ResourceSpecGetter{}
+	nicSpecs = append(nicSpecs, m.PrimaryNICSpec())
```

Comment:
we shouldn't need two functions for primary vs secondary, This function should only range over `m.AzureMachine.Spec.NetworkInterfaces` and pass through the fields to convert them in a `[]*networkinterfaces.NICSpec`. There is still way too much business logic happening in here.

Reply:
I'm happy to combine them back into a single function with just one loop. Working on that now. 

Regarding the business logic, that is a more difficult question. The bulk of it predates this PR. That logic either needs to exist here, move somewhere else, or the features provided be removed or redesigned. Those problems are all unrelated to this PR though.

I am going to attempt to trim down the business logic I added now that we have the defaulting webhook logic in place.